### ISSUE TITLE
[bug] crashes if server does only support client cert

### ISSUE THREAD

**Issue by TuxCoder** [iNPUTmice/Conversations#3723]
#### General information

* **Version:** 2.8.1+fcr (fdroid)
* **Device:** Xiaomi Mi xx, and others
* **Android Version:**  Android 9 (stock)
* **Server name:** `jabber (d0t) o-g (d0t)` , self hosted client cert only
* **Server software:**  prosody 0.11.2
* **Installed server modules:** Stream Managment, CSI, MAM,...
* **Conversations source:** F-Droid

#### Steps to reproduce

1. try to add account to that host above (randome name/password)

#### Expected result
show error/ switch to client cert auth

#### Actual result

crash

#### Debug output

````
05-12 14:28:08.599 29412 29438 E AndroidRuntime: FATAL EXCEPTION: Thread-4
05-12 14:28:08.599 29412 29438 E AndroidRuntime: Process: eu.siacs.conversations, PID: 29412
05-12 14:28:08.599 29412 29438 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
05-12 14:28:08.599 29412 29438 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.authenticate(XmppConnection.java:902)
05-12 14:28:08.599 29412 29438 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.processStreamFeatures(XmppConnection.java:850)
05-12 14:28:08.599 29412 29438 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.processStream(XmppConnection.java:466)
05-12 14:28:08.599 29412 29438 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.switchOverToTls(XmppConnection.java:807)
05-12 14:28:08.599 29412 29438 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.processStream(XmppConnection.java:468)
05-12 14:28:08.599 29412 29438 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.connect(XmppConnection.java:376)
05-12 14:28:08.599 29412 29438 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.run(XmppConnection.java:455)
05-12 14:28:08.599 29412 29438 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:764)

````

**Comment by iNPUTmice** [comment]
This has been fixed in e4b906ebebe196dcabf2341a9f55a699aa39ca5d (2.8.2+)

**Comment by TuxCoder** [comment]
nice that was a fast fix :)
Thank you very much!
Shows now an error.
